runtime/src/cache_hash_data.rs

```diff
@@ -62,51 +67,38 @@ impl CacheHashDataFile {
         stats.decode_us += m2.as_us();
     }
 
-    /// get '&mut EntryType' from cache file [ix]
-    fn get_mut(&mut self, ix: u64) -> &mut EntryType {
-        let item_slice = self.get_slice_internal(ix);
-        unsafe {
-            let item = item_slice.as_ptr() as *mut EntryType;
-            &mut *item
-        }
-    }
-
     /// get '&[EntryType]' from cache file [ix..]
     fn get_slice(&self, ix: u64) -> &[EntryType] {
         let start = self.get_element_offset_byte(ix);
         let item_slice: &[u8] = &self.mmap[start..];
-        let remaining_elements = item_slice.len() / std::mem::size_of::<EntryType>();
+        let remaining_elements = item_slice.len() / CELL_SIZE;
         unsafe {
             let item = item_slice.as_ptr() as *const EntryType;
             std::slice::from_raw_parts(item, remaining_elements)
         }
     }
 
+    /// get '&mut [EntryType]' from cache file [ix..]
+    fn get_slice_mut(&self, ix: u64) -> &mut [EntryType] {
```

Comment:
I see you're still working on it and I'll do a proper review when you're done. I'll point out that this is undefined behaviour tho. You need to make `get_slice_mut` take `&mut self` or it becomes possible to have aliasing mut references which is UB.